blue-green-deploy/blue-green-deploy.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: blue-green-deploy
+spec:
+  workspaces:
+    - name: manifest-dir
+      description: Consisting of kubernetes manifests
+    - name: kubeconfig-dir
+      description: Can be used in case kubeconfig file is mounted
+  params:
+    - name: SERVICE_NAME
+      type: string
+      description: Name of the service which is pointing to existing deployment
+    - name: NEW_VERSION
+      type: string
+      description: The version of newer deployment which is to be patched to existing service
+    - name: MANIFEST
+      type: string
+      default: "."
+      description: The newer version of the deployment to be deployed in another zone
+  steps:
+    - name: change-deployment
+      image: quay.io/openshift/origin-cli:latest
+      workingDir: $(workspaces.manifest-dir.path)
+      script: |
+        #!/usr/bin/env bash
+
+        if [[ -f $(workspaces.kubeconfig-dir.path)/kubeconfig ]]; then
+          export KUBECONFIG=$(workspaces.kubeconfig-dir.path)/kubeconfig
+        fi
+
+        getReadyStatus() {
+          local READY
+          i=0
+          for f in $(kubectl get deploy $1 -o jsonpath='{range .status.conditions[*]}{.reason}{"\n"}{end}')
+          do
+            if [[ $f == "MinimumReplicasAvailable" ]]; then
+              READY=$(kubectl get deploy $1 -o jsonpath='{range .status.conditions['$i']}{.status}{"\n"}{end}')
+              echo $READY
+              return
+            fi
+            ((i++))
```

Comment:
- do we need this variable `i`.

- instead of `echo $READY`, let us do the check `"$READY" != "True"` here and use `return 1` or `return 0`

Reply:
using the variable `i` I am searching for `MinimumReplicasAvailable` in the `JSONArray`. :sweat_smile: . sample JSON is as follows:-
```json
"status": {
        "conditions": [
            {
                "lastTransitionTime": "2020-07-08T05:16:18Z",
                "lastUpdateTime": "2020-07-08T05:16:18Z",
                "message": "Deployment does not have minimum availability.",
                "reason": "MinimumReplicasUnavailable",
                "status": "False",
                "type": "Available"
            },
            {
                "lastTransitionTime": "2020-07-08T05:16:18Z",
                "lastUpdateTime": "2020-07-08T05:16:18Z",
                "message": "ReplicaSet \"myapp-v1-69cbc9bd59\" is progressing.",
                "reason": "ReplicaSetUpdated",
                "status": "True",
                "type": "Progressing"
            }
        ],
        "observedGeneration": 1,
        "replicas": 3,
        "unavailableReplicas": 3,
        "updatedReplicas": 3
    }
```